src/entity/projectile/WindCharge.php

```diff
@@ -0,0 +1,227 @@
+use pocketmine\math\AxisAlignedBB;
+use pocketmine\math\RayTraceResult;
+use pocketmine\math\Vector3;
+use pocketmine\nbt\tag\CompoundTag;
+use pocketmine\network\mcpe\protocol\types\entity\EntityIds;
+use pocketmine\world\particle\WindExplosionParticle;
+use pocketmine\world\sound\DoorSound;
+use pocketmine\world\sound\FlintSteelSound;
+use pocketmine\world\sound\RedstonePowerOffSound;
+use pocketmine\world\sound\RedstonePowerOnSound;
+use pocketmine\world\sound\WindChargeBurstSound;
+
+use function ceil;
+use function floor;
+use function round;
+use function time;
+
+class WindCharge extends Throwable{
+	protected float $radius = 2.5;
+	protected float $damage = 0.5;
+
+	protected float $createdAt = 0;
+	protected float $lifetime = 300;
+
+	public static function getNetworkTypeId() : string{ return EntityIds::WIND_CHARGE_PROJECTILE; }
+
+	protected function getInitialDragMultiplier() : float{ return 0; }
+	protected function getInitialGravity() : float{ return 0; }
+
+	protected function initEntity(CompoundTag $nbt) : void{
+		parent::initEntity($nbt);
+
+		$this->createdAt = time();
+	}
+
+	public function attack(EntityDamageEvent $source) : void{
+		if(!$source instanceof EntityDamageByEntityEvent) return;
+
+		if(($entity = $source->getDamager()) == null) return;
+		$this->setOwningEntity($entity);
+
+		$this->setMotion($entity->getDirectionVector()->multiply(1.5));
+	}
+
+	protected function onHitEntity(Entity $entityHit, RayTraceResult $hitResult) : void{
+		if($this->getOwningEntity() === null)$ev = new EntityDamageByEntityEvent($this, $entityHit, EntityDamageEvent::CAUSE_PROJECTILE, $this->damage);
+		else $ev = new EntityDamageByChildEntityEvent($this->getOwningEntity(), $this, $entityHit, EntityDamageEvent::CAUSE_PROJECTILE, $this->damage);
+		$entityHit->attack($ev);
+
+		$this->flagForDespawn();
+	}
+
+	protected function onHitBlock(Block $blockHit, RayTraceResult $hitResult) : void{
+		parent::onHitBlock($blockHit, $hitResult);
+		$bound = $this->getBound($blockHit->getPosition(), $this->radius - 1);
+
+		for($x = $bound->minX; $x <= $bound->maxX; $x++) {
+			for($y = $bound->minY; $y <= $bound->maxY; $y++) {
+				for($z = $bound->minZ; $z <= $bound->maxZ; $z++) {
+					$affectedBlock = $this->getWorld()->getBlockAt((int) floor($x), (int) floor($y), (int) floor($z));
+
+					$this->checkAffect($affectedBlock);
+				}
+			}
+		}
+	}
+
+	protected function entityBaseTick(int $tickDiff = 1) : bool{
+		parent::entityBaseTick($tickDiff);
+
+		if($this->createdAt + $this->lifetime <= time()) $this->flagForDespawn();
+
+		return true;
+	}
+
+	protected function onHit(ProjectileHitEvent $event) : void{
+		$source = $this->getLocation();
+
+		$this->getWorld()->addSound($event->getRayTraceResult()->getHitVector(), new WindChargeBurstSound());
+		$this->getWorld()->addParticle($source, new WindExplosionParticle());
+
+		$list = $source->getWorld()->getNearbyEntities($this->getBound($source, $this->radius));
```

Comment:
Use `World::getCollidingEntities()` instead of `getNearbyEntities` since this behavior should check if the entity is colliderable. Also pass `$this` as the `entity` algument, so this way you can be sure that this entity is not on the list.